common/Environments/Helpers/EnvironmentsNotificationHelper.cs

```diff
@@ -214,37 +226,64 @@ private void AddUserToHyperVAdminGroupAndEnableHyperV(Notification notification)
+                        LogEnablementTelemetry(telemetryEnablementMap);
                         ShowRestartNotification();
                         return;
                     case 2:
                         // Hyper-V Feature is already enabled and the script successfully added the user to the Hyper-V Admin group.
                         _shouldShowHyperVRebootButton = true;
+                        telemetryEnablementMap.Add(FeatureEnablementKind.HyperVAdminGroup, EnvironmentsTelemetryStatus.Succeeded);
+                        LogEnablementTelemetry(telemetryEnablementMap);
                         ShowRestartNotification();
                         return;
                     case 3:
                         // Hyper-V Feature is already enabled and the script failed to add the user to the Hyper-V Admin group.
+                        telemetryEnablementMap.Add(FeatureEnablementKind.HyperVAdminGroup, EnvironmentsTelemetryStatus.Failed);
+                        LogEnablementTelemetry(telemetryEnablementMap, "Failed to enable Hyper-V Feature: exit code 3");
                         ShowErrorWithRebootAfterExecutionMessage(_stringResource.GetLocalized("UserNotAddedToHyperVAdminGroupMessage"));
                         return;
                     case 4:
                         // The user is already in the Hyper-V Admin group and the script successfully enabled the Hyper-Feature.
+                        telemetryEnablementMap.Add(FeatureEnablementKind.HyperVFeature, EnvironmentsTelemetryStatus.Succeeded);
+                        LogEnablementTelemetry(telemetryEnablementMap);
                         _shouldShowHyperVRebootButton = true;
                         ShowRestartNotification();
                         return;
                     case 5:
                         // The user is already in the Hyper-V Admin group and the script failed to enable the Hyper-Feature.
+                        telemetryEnablementMap.Add(FeatureEnablementKind.HyperVFeature, EnvironmentsTelemetryStatus.Failed);
+                        LogEnablementTelemetry(telemetryEnablementMap, "Failed to enable Hyper-V Feature: exit code 5");
                         ShowErrorWithRebootAfterExecutionMessage(_stringResource.GetLocalized("UnableToEnableHyperVFeatureMessage"));
                         return;
                     case 6:
-                        // Display nothing as there is no work to be done
+                        // Display nothing as there is no work to be done, as the feature is enabled and user is in the group.
+                        telemetryEnablementMap.Add(FeatureEnablementKind.HyperVFeature, EnvironmentsTelemetryStatus.NoOperation);
+                        telemetryEnablementMap.Add(FeatureEnablementKind.HyperVAdminGroup, EnvironmentsTelemetryStatus.NoOperation);
+                        LogEnablementTelemetry(telemetryEnablementMap);
```

Comment:
`LogEnablementTelemetry` can be moved out of the switch because it is used in every case.

Reply:
will update, thanks